### PR TITLE
add dockerfile

### PR DESCRIPTION
**Pull request by coryschwartz**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+FROM  golang:alpine as builder
+WORKDIR /build
+COPY go.mod go.sum ./
+RUN go mod download
```
Comment (marten-seemann) [inline Dockerfile:4]: I don't think you need this, `go build` will download dependencies.

**Review by marten-seemann — CHANGES_REQUESTED** [review]
This doesn't build the current master, this builds the latest release.

File: Dockerfile (hunk 1 above)
Comment (coryschwartz) [reply Dockerfile:4]: True.

The reason I did this was to make the build a little faster. download on the first run, and then subsequent runs would be able to use this docker layer.

Comment (coryschwartz) [reply Dockerfile]: Edited, so it doesn't run go mod download.

**Comment by coryschwartz** [comment]
@marten-seemann Does this look okay now?

For context, I'm intending to use this on kubernetes with with IPFS nodes to do hole punching using this.

**Comment by hsanjuan** [comment]
> This doesn't build the current master, this builds the latest release.

I don't understand this comment. I think it worked fine with `go mod download`? This is a common pattern ([i.e.](https://petomalina.medium.com/using-go-mod-download-to-speed-up-golang-docker-builds-707591336888))

**Review by marten-seemann — APPROVED** [review]

**Pull request merged**